core/src/main/java/io/undertow/server/handlers/resource/URLResource.java

```diff
@@ -88,7 +89,13 @@ private void openConnection() {
                     contentLength = null;
                     return;
                 }
-                lastModified =  new Date(connection.getLastModified());
+                if (url.getProtocol().equals("jar")) {
+                    connection.setUseCaches(false);
+                    URL jar = ((JarURLConnection) connection).getJarFileURL();
+                    lastModified = new Date(new File(jar.getFile()).lastModified());
```

Comment:
Jar files can contain last modified information, so the last modified time of the jar itself may not actually be the last modified time.

Reply:
Looks like JarURLConnection does not take this into account anyway